### PR TITLE
feat(signed-urls): don't rely on access in token, hit db

### PR DESCRIPTION
**Pull request by Avantol13**
Description about what this pull request does.

Please make sure to follow the [DEV guidelines](https://gen3.org/resources/developer/dev-introduction/) before asking for review.

### New Features
- rely on db for signed url authz checks, don't rely on incomplete token information

### Improvements
- truncate user access in token instead of completely removing

### CODE REVIEW

File: fence/jwt/token.py [hunk 1]
```diff
@@ -381,6 +382,11 @@ def generate_signed_access_token(
     if len(dict(user.project_access)) < config["TOKEN_PROJECTS_CUTOFF"]:
         claims["context"]["user"]["projects"] = dict(user.project_access)
     else:
+        # truncate to configured number of projects in token
+        claims["context"]["user"]["projects"] = {
+            k: dict(user.project_access)[k]
+            for k in list(dict(user.project_access))[: config["TOKEN_PROJECTS_CUTOFF"]]
```
Comment (paulineribeyre) [inline fence/jwt/token.py]: i found this which i think is easier to read and would let you remove the if/else
```
import itertools
n = config["TOKEN_PROJECTS_CUTOFF"]
truncated_projects = dict(itertools.islice(user.project_access.items(), n))
claims["context"]["user"]["projects"] = truncated_projects
```

File: fence/jwt/token.py [hunk 2]
```diff
@@ -367,6 +367,7 @@ def generate_signed_access_token(
                 "name": user.username,
                 "is_admin": user.is_admin,
                 "google": {"proxy_group": user.google_proxy_group_id},
+                "projects": {},
```
Comment (paulineribeyre) [inline fence/jwt/token.py]: i don't think you need this

Comment (paulineribeyre) [reply fence/jwt/token.py]: actually we could just do `"projects": truncated_projects ` here

File: fence/jwt/token.py (hunk 1 above)
Comment (Avantol13) [reply fence/jwt/token.py]: I did see that as an option, my opinion is that the dictionary comprehension is a bit clearer but I suppose that's up for debate. I could remove the if else with the current version too actually... :thinking:

Comment (paulineribeyre) [reply fence/jwt/token.py]:
```
projects = dict(user.project_access)
for key in list(projects.keys())[config["TOKEN_PROJECTS_CUTOFF"]:]
  del projects[key]
```
maybe?

File: fence/jwt/token.py [hunk 3]
```diff
@@ -381,9 +381,18 @@ def generate_signed_access_token(
     if len(dict(user.project_access)) < config["TOKEN_PROJECTS_CUTOFF"]:
         claims["context"]["user"]["projects"] = dict(user.project_access)
     else:
+        # truncate to configured number of projects in token
+        projects = dict(user.project_access)
+        for key in list(projects.keys())[1:]:
```
Comment (paulineribeyre) [inline fence/jwt/token.py]: why 1?
```suggestion
        for key in list(projects)[config["TOKEN_PROJECTS_CUTOFF"]:]:
```

Comment (Avantol13) [reply fence/jwt/token.py]: Whoops, applied your fix

**Pull request merged**